### PR TITLE
Add a table which holds the list of Repos which are Dataset

### PR DESCRIPTION
**Pull request by anandjakhaniya**

### CODE REVIEW

File: backend/src/main/java/ai/verta/modeldb/entities/versioning/DatasetRepositoryMappingEntity.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    this.isDatasetRepository = isDatasetRepository;
+  }
+
+  @Id
+  @OneToOne(cascade = CascadeType.ALL)
+  @JoinColumn(name = "repository_id")
+  private RepositoryEntity repositoryEntity;
+
+  @Column(name = "is_dataset_repository")
+  private Boolean isDatasetRepository = false;
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/entities/versioning/DatasetRepositoryMappingEntity.java]: is the second column required?
we can only store the repos which are dataset , why store rows which have this as false?

File: backend/src/main/java/ai/verta/modeldb/entities/versioning/RepositoryEntity.java [hunk 2]
```diff
@@ -231,7 +239,6 @@ private void updateAttribute(List<KeyValue> attributes) throws InvalidProtocolBu
   }
 
   public boolean isDataset() {
-    // TODO: keep track if it is a dataset https://vertaai.atlassian.net/browse/VR-4571
-    return true;
+    return datasetRepositoryMappingEntity.getIsDatasetRepository();
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/entities/versioning/RepositoryEntity.java]: return datasetRepositoryMappingEntity!= null 

would this work?

**Review by ravishetye — CHANGES_REQUESTED** [review]
can merge if comment are resolved as suggested

File: backend/src/main/java/ai/verta/modeldb/entities/versioning/DatasetRepositoryMappingEntity.java (hunk 1 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/entities/versioning/DatasetRepositoryMappingEntity.java]: Fixed

File: backend/src/main/java/ai/verta/modeldb/entities/versioning/RepositoryEntity.java (hunk 2 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/entities/versioning/RepositoryEntity.java]: Yes, it works. I have changed it as per your suggestion

**Pull request merged**